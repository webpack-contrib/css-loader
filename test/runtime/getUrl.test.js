/**
 * @jest-environment jsdom
 */

const getUrl = require("../../src/runtime/getUrl");

describe("escape", () => {
  it("should escape url", () => {
    expect(getUrl(true)).toMatchSnapshot();
    expect(getUrl(null)).toMatchSnapshot();
    // eslint-disable-next-line no-undefined
    expect(getUrl(undefined)).toMatchSnapshot();
    expect(getUrl("image.png")).toMatchSnapshot();
    expect(getUrl('"image.png"')).toMatchSnapshot();
    expect(getUrl("'image.png'")).toMatchSnapshot();
    expect(getUrl("image other.png")).toMatchSnapshot();
    expect(getUrl('"image other.png"')).toMatchSnapshot();
    expect(getUrl("'image other.png'")).toMatchSnapshot();
    expect(getUrl('image"other.png')).toMatchSnapshot();
    expect(getUrl("image\nother.png")).toMatchSnapshot();

    expect(getUrl("image.png", { hash: "#hash" })).toMatchSnapshot();
    expect(getUrl('"image.png"', { hash: "#hash" })).toMatchSnapshot();
    expect(getUrl("'image.png'", { hash: "#hash" })).toMatchSnapshot();
    expect(getUrl("image other.png", { hash: "#hash" })).toMatchSnapshot();
    expect(getUrl('"image other.png"', { hash: "#hash" })).toMatchSnapshot();
    expect(getUrl("'image other.png'", { hash: "#hash" })).toMatchSnapshot();

    expect(getUrl("image other.png", { needQuotes: true })).toMatchSnapshot();
    expect(getUrl("'image other.png'", { needQuotes: true })).toMatchSnapshot();
    expect(getUrl('"image other.png"', { needQuotes: true })).toMatchSnapshot();

    expect(
      getUrl({ default: "image.png", __esModule: true })
    ).toMatchSnapshot();
    expect(
      getUrl({ default: "'image.png'", __esModule: true })
    ).toMatchSnapshot();
    expect(
      getUrl({ default: '"image.png"', __esModule: true })
    ).toMatchSnapshot();
    expect(
      getUrl({ default: "image other.png", __esModule: true })
    ).toMatchSnapshot();
    expect(
      getUrl({ default: '"image other.png"', __esModule: true })
    ).toMatchSnapshot();
    expect(
      getUrl({ default: "'image other.png'", __esModule: true })
    ).toMatchSnapshot();
    expect(
      getUrl({ default: 'image"other.png', __esModule: true })
    ).toMatchSnapshot();
    expect(
      getUrl({ default: "image\nother.png", __esModule: true })
    ).toMatchSnapshot();

    expect(
      getUrl({ default: "image.png", __esModule: true }, { hash: "#hash" })
    ).toMatchSnapshot();
    expect(
      getUrl({ default: '"image.png"', __esModule: true }, { hash: "#hash" })
    ).toMatchSnapshot();
    expect(
      getUrl({ default: "'image.png'", __esModule: true }, { hash: "#hash" })
    ).toMatchSnapshot();
    expect(
      getUrl(
        { default: "image other.png", __esModule: true },
        { hash: "#hash" }
      )
    ).toMatchSnapshot();

    expect(
      getUrl(
        { default: "image other.png", __esModule: true },
        { needQuotes: true }
      )
    ).toMatchSnapshot();
    expect(
      getUrl(
        { default: "'image other.png'", __esModule: true },
        { needQuotes: true }
      )
    ).toMatchSnapshot();
    expect(
      getUrl(
        { default: '"image other.png"', __esModule: true },
        { needQuotes: true }
      )
    ).toMatchSnapshot();

    expect(
      getUrl("image other.png", { hash: "#hash", needQuotes: true })
    ).toMatchSnapshot();
    expect(
      getUrl('"image other.png"', { hash: "#hash", needQuotes: true })
    ).toMatchSnapshot();
    expect(
      getUrl("'image other.png'", { hash: "#hash", needQuotes: true })
    ).toMatchSnapshot();
    expect(
      getUrl("image other.png", { hash: "#hash", needQuotes: true })
    ).toMatchSnapshot();

    expect(
      getUrl(
        { default: "image other.png", __esModule: true },
        { hash: "#hash", needQuotes: true }
      )
    ).toMatchSnapshot();
    expect(
      getUrl(
        { default: '"image other.png"', __esModule: true },
        { hash: "#hash", needQuotes: true }
      )
    ).toMatchSnapshot();
    expect(
      getUrl(
        { default: "'image other.png'", __esModule: true },
        { hash: "#hash", needQuotes: true }
      )
    ).toMatchSnapshot();
    expect(
      getUrl(
        { default: "image other.png", __esModule: true },
        { hash: "#hash", needQuotes: true }
      )
    ).toMatchSnapshot();
  });
});
