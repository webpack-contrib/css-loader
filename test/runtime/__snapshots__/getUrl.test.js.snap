// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`escape should escape url 1`] = `"true"`;

exports[`escape should escape url 2`] = `null`;

exports[`escape should escape url 3`] = `undefined`;

exports[`escape should escape url 4`] = `"image.png"`;

exports[`escape should escape url 5`] = `"image.png"`;

exports[`escape should escape url 6`] = `"image.png"`;

exports[`escape should escape url 7`] = `"\\"image other.png\\""`;

exports[`escape should escape url 8`] = `"\\"image other.png\\""`;

exports[`escape should escape url 9`] = `"\\"image other.png\\""`;

exports[`escape should escape url 10`] = `"\\"image\\\\\\"other.png\\""`;

exports[`escape should escape url 11`] = `"\\"image\\\\nother.png\\""`;

exports[`escape should escape url 12`] = `"image.png#hash"`;

exports[`escape should escape url 13`] = `"image.png#hash"`;

exports[`escape should escape url 14`] = `"image.png#hash"`;

exports[`escape should escape url 15`] = `"\\"image other.png#hash\\""`;

exports[`escape should escape url 16`] = `"\\"image other.png#hash\\""`;

exports[`escape should escape url 17`] = `"\\"image other.png#hash\\""`;

exports[`escape should escape url 18`] = `"\\"image other.png\\""`;

exports[`escape should escape url 19`] = `"\\"image other.png\\""`;

exports[`escape should escape url 20`] = `"\\"image other.png\\""`;

exports[`escape should escape url 21`] = `"image.png"`;

exports[`escape should escape url 22`] = `"image.png"`;

exports[`escape should escape url 23`] = `"image.png"`;

exports[`escape should escape url 24`] = `"\\"image other.png\\""`;

exports[`escape should escape url 25`] = `"\\"image other.png\\""`;

exports[`escape should escape url 26`] = `"\\"image other.png\\""`;

exports[`escape should escape url 27`] = `"\\"image\\\\\\"other.png\\""`;

exports[`escape should escape url 28`] = `"\\"image\\\\nother.png\\""`;

exports[`escape should escape url 29`] = `"image.png#hash"`;

exports[`escape should escape url 30`] = `"image.png#hash"`;

exports[`escape should escape url 31`] = `"image.png#hash"`;

exports[`escape should escape url 32`] = `"\\"image other.png#hash\\""`;

exports[`escape should escape url 33`] = `"\\"image other.png\\""`;

exports[`escape should escape url 34`] = `"\\"image other.png\\""`;

exports[`escape should escape url 35`] = `"\\"image other.png\\""`;

exports[`escape should escape url 36`] = `"\\"image other.png#hash\\""`;

exports[`escape should escape url 37`] = `"\\"image other.png#hash\\""`;

exports[`escape should escape url 38`] = `"\\"image other.png#hash\\""`;

exports[`escape should escape url 39`] = `"\\"image other.png#hash\\""`;

exports[`escape should escape url 40`] = `"\\"image other.png#hash\\""`;

exports[`escape should escape url 41`] = `"\\"image other.png#hash\\""`;

exports[`escape should escape url 42`] = `"\\"image other.png#hash\\""`;

exports[`escape should escape url 43`] = `"\\"image other.png#hash\\""`;
